common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -169,6 +239,46 @@ private static boolean matchAt(final UTF8String target, final UTF8String pattern
         pos, pos + pattern.numChars()), pattern, collationId).last() == 0;
     }
 
+    private static int findInSet(UTF8String match, UTF8String set, int collationId) {
+      if (match.contains(UTF8String.fromString(","))) {
+        return 0;
+      }
+
+      StringSearch stringSearch = CollationFactory.getStringSearch(set, match, collationId);
+
+      String setString = set.toString();
```

Comment:
I think CollationFactory.getStringSearch(set, match, collationId) already does set.toString()

there's no need to do that twice, let's add:

  public static StringSearch getStringSearch(
      final String targetString,
      final String pattern,
      final int collationId) {
    CharacterIterator target = new StringCharacterIterator(targetString);
    Collator collator = CollationFactory.fetchCollation(collationId).collator;
    return new StringSearch(pattern, target, (RuleBasedCollator) collator);
  }
to CollationFactory

this way, we can get the stringSearch and setString instances here, without extra String allocations

Reply:
and if you do that, I'd say that the UTF8String implementation for `getStringSearch` can amount to:

```
  public static StringSearch getStringSearch(
      final UTF8String left,
      final UTF8String right,
      final int collationId) {
    return getStringSearch(left.toString(), right.toString(), collationId);
  }
```

this way, we can have a more complete API

Reply:
Added